spec/queries/scrape_result_search_query_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe(ScrapeResultSearchQuery) do
+  describe '#call' do
+    context 'given :exact query type' do
+      it 'returns all URLs satisfy the query pattern grouped by their keyword_id' do
+        url = 'https://google.com'
+        keyword = Fabricate(:parsed_keyword) { keyword_search_entries(count: 0) }
+        Fabricate.times(2, :keyword_search_entry, urls: [url], keyword_id: keyword.id)
+        Fabricate(:keyword_search_entry, urls: ['https://ruby-lang.org'], keyword_id: keyword.id)
+
+        result = described_class.new(query_type: :exact, pattern: url).call
+
+        expect(result).to(contain_exactly({ keyword_id: keyword.id, urls: [url, url] }))
+      end
+    end
+
+    context 'given :partial query type' do
+      it 'returns all URLs satisfy the query pattern grouped by their keyword_id' do
+        url_a = 'https://google.com'
+        url_b = 'https://www.google.com'
+        keyword = Fabricate(:parsed_keyword) { keyword_search_entries(count: 0) }
+        Fabricate(:keyword_search_entry, urls: [url_a], keyword_id: keyword.id)
+        Fabricate(:keyword_search_entry, urls: [url_b], keyword_id: keyword.id)
+        Fabricate(:keyword_search_entry, urls: ['https://ruby-lang.org'], keyword_id: keyword.id)
+
+        result = described_class.new(query_type: :partial, pattern: 'google').call
+
+        expect(result).to(contain_exactly({ keyword_id: keyword.id, urls: [url_a, url_b] }))
+      end
+    end
+
+    context 'given :pattern query type' do
+      it 'returns all URLs satisfy the query pattern grouped by their keyword_id' do
+        url_a = 'https://google.com'
+        url_b = 'https://www.google.com'
+        keyword = Fabricate(:parsed_keyword) { keyword_search_entries(count: 0) }
+        Fabricate(:keyword_search_entry, urls: [url_a], keyword_id: keyword.id)
+        Fabricate(:keyword_search_entry, urls: [url_b], keyword_id: keyword.id)
+        Fabricate(:keyword_search_entry, urls: ['https://ruby-lang.org'], keyword_id: keyword.id)
+
+        result = described_class.new(query_type: :pattern, pattern: 'go{2}\w+\.com$').call
+
+        expect(result).to(contain_exactly({ keyword_id: keyword.id, urls: [url_a, url_b] }))
+      end
+    end
+
+    context 'given EMPTY pattern' do
```

Comment:
Shouldn't these two last tests be in a separate context group, as these are called at the initialize method and complimented with another test?

```ruby
describe '#new' do
  context 'given valid params' do
     it 'does NOT raise errors' do
       #...
     end 
   end 

   context 'given invalid params' do
     context 'given EMPTY pattern' do
       #...
     end 

     context 'given INVALID query_type' do
       #...
     end
   end 
end
```

This is more about having tests self-documenting the implementation and making it easier to understand the implementation

Reply:
Agree. These test cases should belong to a different context. Fixed in f815bca0cc57fccf90353111442612ba1ab7bdc4